health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -50,65 +49,55 @@
   protected Result check() throws Exception {
     try {
-      ServiceReference<DataStoreFactory> dataStoreFactoryReference =
-          context.getServiceReference(DataStoreFactory.class);
-      if (dataStoreFactoryReference != null) {
-        DataStoreFactory dataStoreFactory = context.getService(dataStoreFactoryReference);
-        try {
-          DataStore dataStore = dataStoreFactory.createInternal();
-          ensureTableExists(dataStore);
-
-          // insert record
-          dataStore
-              .execute(
-                  dataStore
-                      .queryBuilder()
-                      .insertInto(KEYSPACE_NAME, TABLE_NAME)
-                      .value(PK_COLUMN_NAME, PK_COLUMN_VALUE)
-                      .value(VALUE_COLUMN_NAME, VALUE_COLUMN_VALUE)
-                      .build()
-                      .bind())
-              .get();
+      DataStore dataStore = dataStoreFactory.createInternal();
 
-          Future<ResultSet> rs =
+      // insert record
+      dataStore
+          .execute(
               dataStore
                   .queryBuilder()
-                  .select()
-                  .column(PK_COLUMN_NAME)
-                  .column(VALUE_COLUMN_NAME)
-                  .from(KEYSPACE_NAME, TABLE_NAME)
+                  .insertInto(KEYSPACE_NAME, TABLE_NAME)
+                  .value(PK_COLUMN_NAME, PK_COLUMN_VALUE)
+                  .value(VALUE_COLUMN_NAME, VALUE_COLUMN_VALUE)
                   .build()
-                  .execute();
+                  .bind())
+          .get();
 
-          Row row = rs.get().one();
-          String pk = row.getString(PK_COLUMN_NAME);
-          String value = row.getString(VALUE_COLUMN_NAME);
+      Future<ResultSet> rs =
+          dataStore
+              .queryBuilder()
+              .select()
+              .column(PK_COLUMN_NAME)
+              .column(VALUE_COLUMN_NAME)
+              .from(KEYSPACE_NAME, TABLE_NAME)
+              .build()
+              .execute();
 
-          if (pk == null || value == null) {
-            return Result.unhealthy("DataStore did not return the proper data.");
-          }
+      Row row = rs.get().one();
+      String pk = row.getString(PK_COLUMN_NAME);
+      String value = row.getString(VALUE_COLUMN_NAME);
 
-          if (pk.equals(PK_COLUMN_VALUE) && value.equals(VALUE_COLUMN_VALUE)) {
-            return Result.healthy("DataStore is operational");
-          } else {
-            return Result.healthy("DataStore did not return the proper data.");
-          }
-        } finally {
-          context.ungetService(dataStoreFactoryReference);
-        }
+      if (pk == null || value == null) {
+        return Result.unhealthy("DataStore did not return the proper data.");
       }
 
-      logger.warn("DataStoreFactory service is not available");
-      return Result.unhealthy("DataStoreFactory service is not available");
-
+      if (pk.equals(PK_COLUMN_VALUE) && value.equals(VALUE_COLUMN_VALUE)) {
```

Comment:
If you flip this around you can get rid of the if block above
```suggestion
      if (PK_COLUMN_VALUE.equals(pk) && VALUE_COLUMN_VALUE.equals(value)) {
```